src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -106,21 +120,33 @@ public Task<bool> IsFullyLoadedAsync(CancellationToken cancellationToken)
                         _lastTimeCalled = new ResettableDelay(delayInMilliseconds: delay, AsynchronousOperationListenerProvider.NullListener);
                         _ = _lastTimeCalled.Task.SafeContinueWith(_ => StatusChanged?.Invoke(this, true), TaskScheduler.Default);
 
-                        return SpecializedTasks.False;
+                        return false;
                     }
 
                     if (_lastTimeCalled.Task.IsCompleted)
                     {
                         _lastTimeCalled = null;
-                        return SpecializedTasks.True;
+                        return true;
                     }
 
                     _lastTimeCalled.Reset();
-                    return SpecializedTasks.False;
+                    return false;
                 }
 
                 // we are using this API for now, until platform provide us new API for prototype
-                return KnownUIContexts.SolutionExistsAndFullyLoadedContext.IsActive ? SpecializedTasks.True : SpecializedTasks.False;
+                return KnownUIContexts.SolutionExistsAndFullyLoadedContext.IsActive;
+            }
+
+            private async Task<bool> IsPartialLoadModeExperimentEnabledAsync(CancellationToken cancellationToken)
```

Comment:
It seems we're doing the caching pattern like this in a bunch of places, can we make the pattern more common? I'm still not sure why we're not just making this IsExperimentEnabledAsync() that just does the GetExperimentationService + IsExperimentEnabled.

Reply:
Some of the callers are not asynchronous.

Reply:
> can we make the pattern more common?

I haven't found a common pattern that doesn't come with a hit to performance, clarity, and/or correctness. It would be simpler to not cache the value, but I wasn't sure which methods were called too frequently to make that viable.